polly/lib/Transform/ScheduleOptimizer.cpp

```diff
@@ -860,7 +867,25 @@ static void runIslScheduleOptimizer(
     SC = SC.set_proximity(Proximity);
     SC = SC.set_validity(Validity);
     SC = SC.set_coincidence(Validity);
+
+    // Save error handling behavior
```

Comment:
Would it be possible to use the IslMaxOperationsGuard helper here?  In addition to the code reuse, the RAII pattern is easier to read.